### PR TITLE
fix(ui): Check type is number when using `>` `<`

### PR DESCRIPTION
**Pull request by scttcper**
typescript v5 doesn't like `string | number` being compared using less/greater than. Check for number in a few places and cast others

### CODE REVIEW

**Comment by priscilawebdev** [comment]
This seems like a regression on their end...do you know if they are looking into it?

File: static/app/views/starfish/components/table.tsx [hunk 1]
```diff
@@ -206,7 +206,12 @@ class _Table extends Component<Props, State> {
 
     const fieldName = getAggregateAlias(field);
     const value = dataRow[fieldName];
-    if (tableMeta[fieldName] === 'integer' && defined(value) && value > 999) {
+    if (
+      tableMeta[fieldName] === 'integer' &&
+      defined(value) &&
+      typeof value === 'number' &&
```
Comment (evanpurkhiser) [inline static/app/views/starfish/components/table.tsx:212]: do we even need defined anymore?

File: static/app/utils/profiling/gl/utils.spec.tsx [hunk 2]
```diff
@@ -182,7 +182,7 @@ describe('createShader', () => {
       shaderSource: jest.fn(),
       compileShader: jest.fn(),
       getShaderParameter: jest.fn().mockImplementation(() => 1),
-      COMPILE_STATUS: 1,
+      COMPILE_STATUS: 1 as any,
```
Comment (scttcper) [inline static/app/utils/profiling/gl/utils.spec.tsx:185]: the type here is like `3142 | undefined` and its a test so i'm just casting it to any cc @JonasBa

File: static/app/views/starfish/components/table.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/views/starfish/components/table.tsx]: no guess not

**Review by evanpurkhiser — APPROVED** [review]

**Comment by priscilawebdev** [comment]
@scttcper correcting my comment...now after reading the description (and not only the title hehe), I understand why we are doing that. it makes sense :)

**Pull request merged**